app/src/main/java/com/example/arc_exapmle/viewModel/MainActivityViewModel.kt

```diff
@@ -1,36 +1,78 @@
 package com.example.arc_exapmle.viewModel
 
-import android.app.Application
-import androidx.lifecycle.AndroidViewModel
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.*
 import com.example.arc_exapmle.note.NoteEntity
 import com.example.arc_exapmle.note.NoteRepository
 import com.example.arc_exapmle.note.NoteUI
+import com.example.arc_exapmle.user.UserEntity
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.launch
+import java.util.*
+import kotlin.collections.ArrayList
 
-class MainActivityViewModel(repository: NoteRepository) : ViewModel() {
-
-    private var repository = repository
-    val allNotes: LiveData<List<NoteEntity>> = repository.getAllNotes()
+class MainActivityViewModel(private var repository: NoteRepository) : ViewModel() {
 
     val toastMutableLiveData: MutableLiveData<String> = MutableLiveData()
 
+    val noteUIListMutableLiveData: MutableLiveData<List<NoteUI>> = MutableLiveData()
+
     fun deleteNote(noteUI: NoteUI) {
 
-        repository.delete(noteUI)
+        viewModelScope.launch {
+
+            repository.delete(noteUI)
+
+        }
 
-        toastMutableLiveData.value = "Note deleted Successfully"
 
     }
 
     fun deleteAllNotes() {
 
-        repository.deleteAllNotes()
 
-        toastMutableLiveData.value = "All notes deleted"
+        viewModelScope.launch {
+
+            repository.deleteAllNotes()
+
+
+
+            toastMutableLiveData.value = "All notes deleted"
+        }
+
+
+    }
+
+    fun getAllNotes() {
+
+        viewModelScope.launch {
+
+
+            repository.getAllNotes().distinctUntilChanged().collect() {
+
+                val noteUIList: MutableList<NoteUI> = ArrayList()
+
+                it.iterator().forEach { noteEntity ->
+
+                    noteUIList.add(
+                        NoteUI(
+                            noteEntity.noteId,
+                            noteEntity.title,
+                            noteEntity.description,
+                            noteEntity.priority,
+                            repository.userId
+                        )
+                    )
+
+                }
```

Comment:
much better now that we have this logic in viewModel good job.
You can leverage from kotlin `List` operators much more in places like this. they make your code much more prettier and most of the time better.
IE:. you may use `.map{}` here instead of doing it yourself manually

Reply:
Also Done!